### ISSUE TITLE
[HIGH PRI] Create version of opam file format parser that can be compiled to JS using BuckleScript.

### ISSUE THREAD

**Issue by jordwalke** [jordwalke/esy#90]
Having the ability to parse opam files directly from JS would allow us to clone opam repos and then build them locally easily without having to create opam switches etc.

Old text:
> As a follow up to [this issue](https://github.com/jordwalke/esy/issues/82), now that `esy` can be used as an npm dependency to perform postinstall builds, we need a lighter weight version of esy to be built that doesn't pull in all of yarn's dependencies. Right now npm `esy` is the total CLI. We should split it into two packages.

> Either:
> - `esybuild` (the smaller builder) and `esy` (the cli)
> or
> - `esy`(the smaller builder) and `esy-cli` (the cli).

> The smaller builder has no install capabilities and so needs nothing related to yarn. It assumes all packages have already been converted over to pure npm and pushed somewhere as pure npm.

**Comment by wokalski** [comment]
> It assumes all packages have already been converted over to pure npm and pushed somewhere as pure npm.

Does this mean that it depends on #81? If yes, does it depend on that issue because esy uses some opam packages and that way `esybuild` could have them in its `package.json`?

**Comment by wokalski** [comment]
For anyone working on this:

>I think they actually can be worked on in parallel, but before there's a dependency there before we can actually roll out the npm packages, yeah.

**Comment by jordwalke** [comment]
Repurposing this github issue.

**Comment by kazcw** [comment]
Is this addressed by jordwalke/esy-issues#98?

**Comment by andreypopp** [comment]
Originally by https://github.com/esy-ocaml/esy-opam but now this is abandoned as we ported esy to native — we use `opam-*` libs as is from opam now.

**Comment by kazcw** [comment]
Here is a BuckleScript builder for the current version of opam-file-format: [kazcw/opam-file-format-bs](https://github.com/kazcw/opam-file-format-bs). I'm not sure what else is needed.

**Comment by kazcw** [comment]
Maybe I'm interpreting "parser" too narrowly. Is this bug for the whole opam-to-npm converter discussed in #81?

**Comment by wokalski** [comment]
@kazcw this issue is obsolete because esy has moved on from compiling to JS.

**Comment by jordwalke** [comment]
Yes, thanks for the reminder. I'll close this. In fact most of the issues are in esy repo now.
